src/shims/panic.rs

```diff
@@ -40,15 +40,11 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     fn handle_miri_start_panic(
         &mut self,
         args: &[OpTy<'tcx, Tag>],
-        unwind: Option<mir::BasicBlock>,
+        unwind: StackPopUnwind,
     ) -> InterpResult<'tcx> {
         let this = self.eval_context_mut();
 
         trace!("miri_start_panic: {:?}", this.frame().instance);
-        // Make sure we only start unwinding when this matches our panic strategy.
-        if this.tcx.sess.panic_strategy() != PanicStrategy::Unwind {
```

Comment:
Why did you remove this?

Reply:
`unwind_to_block()` will check this now: https://github.com/rust-lang/rust/blob/1969c2e312303faa458cf19dad9783165e3c72c8/compiler/rustc_middle/src/ty/layout.rs#L2588